purchases/src/main/kotlin/com/revenuecat/purchases/Offerings.kt

```diff
@@ -43,10 +44,20 @@ data class Offerings internal constructor(
         val showNoOffering = placements.offeringIdsByPlacement.containsKey(placementId)
         val offering = offeringForPlacement ?: if (showNoOffering) null else fallbackOffering
 
-        return offering?.withPlacement(placementId)
+        return offering?.withPresentedContext(placementId = placementId, targeting = this.targeting)
     }
 }
 
+/**
+ * This class contains information about the targeting that presented the offering.
+ * @property revision The revision of the targeting that presented the offering.
+ * @property ruleId The rule id of the targeting that presented the offering.
+ */
+internal data class Targeting(
```

Comment:
I'm slightly concerned this name might be too generic... Maybe we could rename `OfferingTargeting`? Or make it an inner class inside `Offering`, so it would be `Offering.Targeting`?

Reply:
I'll rename this to `TargetedOffering` which will match the naming of the backend 👍